Particle Enums: Use Consistently and Rename

ax3l opened the pull request:
- rename: `m_qm` to `qm` because it is charge over mass, not its inverse
- rename AoS to use s-based description by default: `z`->`t`
- Rename `AddNParticles` parameter, since we add particles for fixed `s`
- Use enums consistently whe we access `.pos` arguments, instead of `0-2`

ax3l committed:
Particle Enums: Use Consistently and Rename

- rename: `m_qm` to `qm` because it is charge over mass, not its inverse
- rename AoS to use s-based description by default: `z`->`t`
- Rename `AddNParticles` parameter, since we add particles for fixed `s`
- Use enums consistently whe we access `.pos` arguments, instead of `0-2`

cemitch99 commented on docs/source/usage/python.rst:
```diff
@@ -167,9 +167,9 @@ Particles
 
-   .. py:method:: add_n_particles(lev, x, y, z, px, py, pyz, qm, bchchg)
+   .. py:method:: add_n_particles(lev, x, y, t, px, py, pyz, qm, bchchg)
 
```
I think "pyz" should be "pt" here.

cemitch99 commented on docs/source/usage/python.rst:
```diff
@@ -178,7 +178,7 @@ Particles
+      :param t: positions as time-of-flight in c*t
       :param px: momentum in x
       :param py: momentum in y
       :param pz: momentum in z
```
We should also update pz -> pt, the momentum conjugate to t.
```suggestion
      :param pt: momentum in t
```

cemitch99 commented on src/particles/ImpactXParticleContainer.H:
```diff
@@ -149,7 +149,7 @@ namespace impactx
+         * @param t positions as time-of-flight in c*t
          * @param px momentum in x
          * @param py momentum in y
          * @param pz momentum in z
```
Here I think pz -> pt momentum in t
```suggestion
         * @param pt momentum in t
```

cemitch99 commented on src/particles/ImpactXParticleContainer.H:
```diff
@@ -160,7 +160,7 @@ namespace impactx
+                       amrex::Vector<amrex::ParticleReal> const & t,
                        amrex::Vector<amrex::ParticleReal> const & px,
                        amrex::Vector<amrex::ParticleReal> const & py,
                        amrex::Vector<amrex::ParticleReal> const & pz,
```
Also pz -> pt here
```suggestion
                       amrex::Vector<amrex::ParticleReal> const & pt,
```

cemitch99 commented on src/particles/ImpactXParticleContainer.cpp:
```diff
@@ -81,7 +81,7 @@ namespace impactx
+                                             amrex::Vector<amrex::ParticleReal> const & t,
                                              amrex::Vector<amrex::ParticleReal> const & px,
                                              amrex::Vector<amrex::ParticleReal> const & py,
                                              amrex::Vector<amrex::ParticleReal> const & pz,
```
pz -> pt
```suggestion
                                             amrex::Vector<amrex::ParticleReal> const & pt,
```

cemitch99 commented on src/particles/ImpactXParticleContainer.cpp:
```diff
@@ -92,7 +92,7 @@ namespace impactx
+        AMREX_ALWAYS_ASSERT(x.size() == t.size());
         AMREX_ALWAYS_ASSERT(x.size() == px.size());
         AMREX_ALWAYS_ASSERT(x.size() == py.size());
         AMREX_ALWAYS_ASSERT(x.size() == pz.size());
```
pz -> pt
```suggestion
        AMREX_ALWAYS_ASSERT(x.size() == pt.size());
```

cemitch99 commented on src/particles/ImpactXParticleContainer.cpp:
```diff
@@ -137,7 +137,7 @@ namespace impactx
         pinned_tile.push_back_real(RealSoA::ux, px);
         pinned_tile.push_back_real(RealSoA::uy, py);
         pinned_tile.push_back_real(RealSoA::pt, pz);
```
Both pt and pz appear here.
```suggestion
        pinned_tile.push_back_real(RealSoA::pt, pt);
```

cemitch99 commented on src/particles/transformation/ToFixedS.H:
```diff
@@ -79,11 +79,11 @@ namespace transformation
-            p.pos(2) = ptf*t/(m_pzd+pt);  // This now represents t.
+            p.pos(RealAoS::t) = ptf * z / (m_pzd + pt);  // This now represents t.
             pt = ptf - ptdf;                    // This now represents pt.
 
```
We should double-check this after the change of notation.

cemitch99 commented on src/particles/transformation/ToFixedT.H:
```diff
@@ -79,11 +79,11 @@ namespace transformation
+            p.pos(RealAoS::t) = pz*t/(m_ptd+pt);  // This now represents z.
             pt = pz - pzd;                     // This now represents pz.
 
             // transform momenta to static units (eg, so that momenta are
```
We should double-check this after the change of notation.

cemitch99 commented on src/python/ImpactXParticleContainer.cpp:
```diff
@@ -44,17 +44,17 @@ void init_impactxparticlecontainer(py::module& m)
              py::arg("lev"),
-             py::arg("x"), py::arg("y"), py::arg("z"),
+             py::arg("x"), py::arg("y"), py::arg("t"),
              py::arg("px"), py::arg("py"), py::arg("pz"),
```
I think pz -> pt here.
```suggestion
             py::arg("px"), py::arg("py"), py::arg("pt"),
```

cemitch99 commented on src/python/ImpactXParticleContainer.cpp:
```diff
@@ -44,17 +44,17 @@ void init_impactxparticlecontainer(py::module& m)
+             ":param t: positions as time-of-flight in c*t\n"
              ":param px: momentum in x\n"
              ":param py: momentum in y\n"
              ":param pz: momentum in z\n"
```
pz -> pt:  momentum conjugate to t
```suggestion
             ":param pt: momentum in t\n"
```

cemitch99 reviewed (commented):
Added mostly comments about the appearance of pz vs pt.

ax3l commented on docs/source/usage/python.rst:
```diff
@@ -167,9 +167,9 @@ Particles
    This class stores particles, distributed over MPI ranks.
 
-   .. py:method:: add_n_particles(lev, x, y, z, px, py, pyz, qm, bchchg)
+   .. py:method:: add_n_particles(lev, x, y, t, px, py, pyz, qm, bchchg)
```
```suggestion
   .. py:method:: add_n_particles(lev, x, y, t, px, py, pt, qm, bchchg)
```

ax3l committed:
Change pt, too

Co-authored-by: Chad Mitchell <46825199+cemitch99@users.noreply.github.com>

ax3l commented on src/particles/transformation/ToFixedS.H:
```diff
@@ -57,9 +57,9 @@ namespace transformation
             using namespace amrex::literals;
```
above: `pt` should be called `pz` here.

ax3l commented on src/particles/transformation/ToFixedT.H:
```diff
@@ -79,11 +79,11 @@ namespace transformation
             // py = py;
-            p.pos(2) = pz*t/(m_ptd+pt);  // This now represents z.
+            p.pos(RealAoS::t) = pz*t/(m_ptd+pt);  // This now represents z.
             pt = pz - pzd;                     // This now represents pz.
```
This one should be called `pz = ...`

cemitch99 commented on src/particles/transformation/ToFixedS.H:
```diff
@@ -79,11 +79,11 @@ namespace transformation
             // transform position and momentum (from fixed t to fixed s)
 
-            p.pos(0) = x - px*t/(m_pzd+pt);
+            p.pos(RealAoS::x) = x - px * z / (m_pzd + pt);
```
```suggestion
            p.pos(RealAoS::x) = x - px * z / (m_pzd + pz);
```

cemitch99 commented on src/particles/transformation/ToFixedS.H:
```diff
@@ -79,11 +79,11 @@ namespace transformation
+            p.pos(RealAoS::x) = x - px * z / (m_pzd + pt);
             // px = px;
-            p.pos(1) = y - py*t/(m_pzd+pt);
+            p.pos(RealAoS::y) = y - py * z / (m_pzd + pt);
```
```suggestion
            p.pos(RealAoS::y) = y - py * z / (m_pzd + pz);
```

cemitch99 commented on src/particles/transformation/ToFixedS.H:
```diff
@@ -79,11 +79,11 @@ namespace transformation
+            p.pos(RealAoS::y) = y - py * z / (m_pzd + pt);
             // py = py;
-            p.pos(2) = ptf*t/(m_pzd+pt);  // This now represents t.
+            p.pos(RealAoS::t) = ptf * z / (m_pzd + pt);  // This now represents t.
```
```suggestion
            p.pos(RealAoS::t) = ptf * z / (m_pzd + pz);  // This now represents t.
```

ax3l committed:
Transforms: Clarify In and Out

cemitch99 commented on src/particles/transformation/ToFixedT.H:
```diff
@@ -79,11 +79,11 @@ namespace transformation
             // py = py;
-            p.pos(2) = pz*t/(m_ptd+pt);  // This now represents z.
+            p.pos(RealAoS::t) = pz*t/(m_ptd+pt);  // This now represents z.
             pt = pz - pzd;                     // This now represents pz.
```
```suggestion
            pz = pzf - pzdf;                     // This now represents pz.
```

cemitch99 replied:
It remains to make sure "pzf" and "pzdf" are defined here, analogously to "ptf" and "ptdf" in ToFixedS.H.

ax3l replied:
What does the `f` stand for?

ax3l replied:
Updated to be in sync: added `f` suffixes.

ax3l committed:
Fancy Alias for Fixed t

ax3l commented:
@cemitch99 I found a way to define alias for `z` and `pz`, which makes the code less confusing :)

ax3l committed:
Spaces

ax3l committed:
ToFixedT: Use Name f-suffix as in ToFixedS

ax3l committed:
More Spaces

ax3l committed:
Docs: Mention s-based first

Mention s-based interpretation first, because its the most commonly
used throughout the code base.

cemitch99 commented on src/particles/transformation/ToFixedS.H:
```diff
@@ -68,30 +68,29 @@ namespace transformation
-            pt = ptf - ptdf;                    // This now represents pt.
+            p.pos(RealAoS::t) = ptf * z / (m_pzd + pz);  // This now represents t.
+            pz = ptf - ptdf;                                   // This now represents pt.
 
```
To clarify:  The same names of the momentum components are used for both the input and output values.  Since "pz" is used here, this notation is consistent with the meaning of the input values, but not with the meaning of the output values.  Was that the intention?

ax3l replied:
Yes, I was not sure how to fix the output values, but instead kept a comment on them and clarified in the API docs. I can introduce alias here, but would not make it clearer, I think.

ax3l replied:
Let me try one more :)

ax3l replied:
Updated, let me know if that looks clearer :)

ax3l committed:
Comments

ax3l committed:
Use New Alias Once More

ax3l committed:
Coord Trafo: Use Aliases for Final Values

cemitch99 reviewed (approved):
Ok, looks good to me.  Thanks!

ax3l commented on src/particles/ImpactXParticleContainer.H:
```diff
@@ -60,20 +65,25 @@ namespace impactx
-            pt,  ///< momentum in z, scaled by the magnitude of the reference momentum [unitless] (at fixed t) OR energy deviation, scaled by speed of light * the magnitude of the reference momentum [unitless] (at fixed s)
-            m_qm, ///< charge to mass ratio, in q_e/m_e (q_e/eV) TODO: rename to qm_m
-            w,   ///< particle weight, unitless
+            ux,  ///< momentum in x, scaled by the magnitude of the reference momentum [unitless] (at fixed s or t)
+            uy,  ///< momentum in y, scaled by the magnitude of the reference momentum [unitless] (at fixed s or t)
```
Follow-up: let's call those px and py, too.

ax3l replied:
In #353

ax3l commented on src/particles/ImpactXParticleContainer.H:
```diff
@@ -39,18 +39,23 @@ namespace impactx
-            z,  ///< position in z [m] (at fixed t) OR time-of-flight ct [m] (at fixed s)
+            x,  ///< position in x [m] (at fixed s OR fixed t)
+            y,  ///< position in y [m] (at fixed s OR fixed t)
+            t,  ///< time-of-flight c*t [m] (at fixed s)
```
`c * time-of-flight`

ax3l replied:
In #353
